### ISSUE TITLE
Variational Diffusion Models

### ISSUE THREAD

**Issue by reyllama** [reyllama/paper-reviews#191]
# TL; DR

- Make noise schedule learnable and Fourier-embed latent noises to obtain SOTA likelihood estimation on images.

# Motivation

- Diffusion models have made impressive advances in terms of perceptual quality, but fall short on likelihood estimation compared to autoregressive baselines.

# Formulation

- Define a variable called SNR (*signal-to-noise ratio*) as follows:
![image](https://user-images.githubusercontent.com/46308407/203715065-68368953-7b8c-4169-9507-892f4399325f.png)

- Parameterize the noise using a monotonic neural network.
![image](https://user-images.githubusercontent.com/46308407/203715262-163c196b-186c-41c6-a514-4d97b7caea58.png)

- Design the monotonic network by composing 3 linear layers whose weights are enforced to remain positive.
- With ***variance-preserving diffusion process***, the other variables can be expressed likewise.
![image](https://user-images.githubusercontent.com/46308407/203715418-214e84a2-579b-4412-9433-6233980b3c87.png)

### Discrete-time Diffusion Models

![image](https://user-images.githubusercontent.com/46308407/203715725-bd621df6-2f36-4d63-8ba8-637900a96508.png)

- This can be simplified by using SNR.
![image](https://user-images.githubusercontent.com/46308407/203715834-92e0f288-8b33-4594-a9c2-9dc9ba747214.png)
![image](https://user-images.githubusercontent.com/46308407/203715869-5aaa8bec-8f13-4cdd-a6b4-6dd454f2ba90.png)

- As discrete denoising steps correspond to upper Riemannian sum w.r.t. a monotonically decreasing function, more steps reduce loss.
![image](https://user-images.githubusercontent.com/46308407/203716030-9b743e7c-6254-4b05-a651-b27db608a736.png)

### Continuous-time Diffusion Models

- This view expresses a diffusion process as stochastic differential equation.

![image](https://user-images.githubusercontent.com/46308407/203716208-6657a1f9-3659-4681-a5e8-d7eeea8f9ab8.png)
![image](https://user-images.githubusercontent.com/46308407/203716233-6dae04d5-b710-49cf-a841-32bfed2434cc.png)

- With continuous-time diffusion process, the noise schedule influences the process only through the minimum and maximum SNR value, as shown below.
![image](https://user-images.githubusercontent.com/46308407/203716362-b8016fc8-8169-4bef-81c2-05e24d5f3068.png)

- Even weighted denoising loss can be seamlessly explained under this framework.
![image](https://user-images.githubusercontent.com/46308407/203716460-8125b260-d92a-42c7-9b9b-d27e908c986f.png)

# Evaluation

![image](https://user-images.githubusercontent.com/46308407/203716483-4c432a28-5cdd-4bff-8d87-057bd86a742e.png)
![image](https://user-images.githubusercontent.com/46308407/203716508-a43420fb-968b-4fbd-85be-5ca7a3225712.png)

- Perceptual quality is so-so, as the model allocates much capacity to matching the ***exact pixel values*** for good likelihood estimation, rather than focusing on the global semantic consistency and coarse texture patterns.

![image](https://user-images.githubusercontent.com/46308407/203716696-0935be21-ddd3-4f29-bec2-d7333e0c9051.png)

# Inspiration

- Noise schedule can be learned to render the framework more effective.
- Fourier features let neural networks learn high frequency details (though I'm not 100% sure it's beneficial in general image modeling settings)
